### PR TITLE
User endpoints tests

### PR DESCRIPTION
**Pull request by MrStashy**

### CODE REVIEW

**Review by homonoviscoding — DISMISSED** [review]

**Review by FBagdeli — DISMISSED** [review]

File: src/controllers/cohort.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ import ERR from '../utils/errors.js'
 export const create = async (req, res) => {
   try {
     const createdCohort = await createCohort()
-
     return sendDataResponse(res, 201, createdCohort)
   } catch (e) {
     console.log('An error occured while trying to create a cohort:', e)
```
Comment (Auenc) [inline src/controllers/cohort.js]: I know this isn't your change, but could we change this console.log to console.error please

File: src/middleware/auth.js [hunk 2]
```diff
@@ -44,7 +44,8 @@ export async function validateAuthentication(req, res, next) {
   }
 
   const decodedToken = jwt.decode(token)
-  const foundUser = await User.findById(decodedToken.userId)
+
+  const foundUser = await User.findById(Number(decodedToken))
```
Comment (Auenc) [inline src/middleware/auth.js]: is decodedToken an ID? if so, can we change the `decodedToken` to something like `userId` to make it clear?

File: tests/api/routes/cohort.spec.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        'TEACHER'
+      )
+      const teacherToken = jwt.sign(teacher.id, process.env.JWT_SECRET)
+
+      const response = await supertest(app)
+        .post('/cohorts')
+        .set('Authorization', `Bearer ${teacherToken}`)
+
+      // expect(response.status).toEqual(201)
+      // expect(response.body.id).not.toEqual(undefined)
```
Comment (Auenc) [inline tests/api/routes/cohort.spec.js]: Should these expect's be commented?

**Review by Auenc — CHANGES_REQUESTED** [review]

File: tests/api/routes/cohort.spec.js (hunk 3 above)
Comment (MrStashy) [reply tests/api/routes/cohort.spec.js]: This is part of someone else's branch that I know has subsequently been updated. I'm not sure on protocol here, but they'er basically an irrelevance. Should I just delete them entirely?

File: src/controllers/cohort.js (hunk 1 above)
Comment (MrStashy) [reply src/controllers/cohort.js]: Done

File: src/middleware/auth.js (hunk 2 above)
Comment (MrStashy) [reply src/middleware/auth.js]: Done

File: tests/api/routes/cohort.spec.js (hunk 3 above)
Comment (Auenc) [reply tests/api/routes/cohort.spec.js]: Yeah if the test is not needed, remove them - if the test is still relevant, then no - best to uncomment them

Comment (MrStashy) [reply tests/api/routes/cohort.spec.js]: Cool, done!

**Review by Auenc — APPROVED** [review]

**Review by tzoltie — APPROVED** [review]

**Pull request merged**